src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -69,27 +69,42 @@ def validate_location(cmd, location):
             break
 
 
-def get_chart_path(registry_path, kube_config, kube_context, helm_client_location):
+def get_chart_path(registry_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks=False):
     # Pulling helm chart from registry
     os.environ['HELM_EXPERIMENTAL_OCI'] = '1'
-    pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location)
+    pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks)
 
     # Exporting helm chart after cleanup
-    chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
+    if for_preonboarding_checks:
+        chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'ConnectPrecheckCharts')
+    else:
+        chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
     try:
         if os.path.isdir(chart_export_path):
             shutil.rmtree(chart_export_path)
     except:
-        logger.warning("Unable to cleanup the azure-arc helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
-    export_helm_chart(registry_path, chart_export_path, kube_config, kube_context, helm_client_location)
+        if for_preonboarding_checks:
+            logger.warning("Unable to cleanup the connect-precheck helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
+        else:
+            logger.warning("Unable to cleanup the azure-arc helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
```

Comment:
this log message can also be a single log line after using the parameter of chart_path_name , you wouldn't need two